index.js

```diff
@@ -149,14 +173,15 @@ function applyAttributes(t, openingElement, componentName, sourceFileName, compo
   }
 
   // Add a stable attribute for the source file name (absent for non-root elements)
-  if(
+  if (
     sourceFileName
+    && !ignoredComponentFromOptions
     && (componentName || ignoredElement === false)
     && openingElement.node.attributes.find(node => {
       if (!node.name) return
       return node.name.name === sourceFileAttributeName
     }
-  ) == null){
+  ) == undefined){
```

Comment:
why this change?

Reply:
Ah good point! I think I had a bug in line 137 where I was strictly checking `!== null` but `find` returns `undefined` when no match.
So I probably overkilled by changing all of the `.find` compares without realizing these aren't type matching. For consistency I can update line 137 to do `!= null` instead